tests/CMakeLists.txt

```
@@ -6,15 +6,24 @@
 # SPDX-License-Identifier: Apache-2.0
 # ~~~
 
-# Test add_subdirectory suppport
-add_test(NAME integration.add_subdirectory
+# Test add_subdirectory suppport for headers
+add_test(NAME integration.add_subdirectory_headers
     COMMAND ${CMAKE_CTEST_COMMAND}
         --build-and-test ${CMAKE_CURRENT_LIST_DIR}/integration
-                         ${CMAKE_CURRENT_BINARY_DIR}/add_subdirectory
+                         ${CMAKE_CURRENT_BINARY_DIR}/add_subdirectory_headers
         --build-generator ${CMAKE_GENERATOR}
         --build-options -DFIND_PACKAGE_TESTING=OFF -DVULKAN_HEADERS_ENABLE_MODULE=OFF
```

I don't see a reason for 2 add_subdirectory tests.

```suggestion
        --build-options -DFIND_PACKAGE_TESTING=OFF -DVULKAN_HEADERS_ENABLE_MODULE=${VULKAN_HEADERS_ENABLE_MODULE}
```

---

I separated them since I did not yet know how to best filter out module support testing for certain CI envs, passing the var like you did here seems like the best solution!